GitUI/Infrastructure/Telemetry/DiagnosticsClient.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+using GitExtUtils.GitUI;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace GitUI.Infrastructure.Telemetry
+{
+    public static class DiagnosticsClient
+    {
+        private static bool _initialized;
+
+        private static TelemetryClient _client;
+
+        public static void Initialize(bool isDirty)
+        {
+            TelemetryConfiguration.Active.InstrumentationKey = "2ef275e3-8850-4305-9d7c-825a60c3d296";
+            TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = Debugger.IsAttached;
+            TelemetryConfiguration.Active.TelemetryInitializers.Add(new AppInfoTelemetryInitializer(isDirty));
+            TelemetryConfiguration.Active.TelemetryInitializers.Add(new DeviceTelemetryInitializer());
+
+            _initialized = true;
+
+            _client = new TelemetryClient();
+
+            var properties = new Dictionary<string, string>
+            {
+                // environment
+                { "DotNetVersion", RuntimeInformation.FrameworkDescription },
+
+                // languages
+                { "CurrentCulture", CultureInfo.CurrentCulture.Name },
+                { "CurrentUICulture", CultureInfo.CurrentUICulture.Name }
+            };
+            AttachMonitorInformation(properties);
+
+            _client.TrackEvent("Startup.Environment", properties);
+        }
+
+        public static void OnExit()
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            _client.Flush();
+
+            // Allow time for flushing:
+            System.Threading.Thread.Sleep(1000);
```

Comment:
This is OK for me since the application disappears at once. It is only visible in the Process Explorer for this additional second.